tests/Test.ADAL.NET.Integration/FederatedUserCredentialTests.cs

```diff
@@ -89,5 +94,208 @@ public void Initialize()
+
+            // All mocks are consumed
+            Assert.AreEqual(0, HttpMessageHandlerFactory.MockHandlersCount());
+        }
+
+
+        [TestMethod]
+        [Description("No password provided and Mex does not return integrated auth endpoint.")]
+        public async Task NoPasswordAndMexEndpointFailsToResolveTestAsync()
+        {
+            AuthenticationContext context = new AuthenticationContext(TestConstants.DefaultAuthorityCommonTenant, new TokenCache());
+            await context.Authenticator.UpdateFromTemplateAsync(null);
+
+            HttpMessageHandlerFactory.AddMockHandler(new MockHttpMessageHandler()
+            {
+                Method = HttpMethod.Get,
+                ResponseMessage = new HttpResponseMessage(HttpStatusCode.OK)
+                {
+                    Content = new StringContent("{\"ver\":\"1.0\",\"account_type\":\"federated\",\"domain_name\":\"microsoft.com\"," +
+                                                "\"federation_protocol\":\"WSTrust\",\"federation_metadata_url\":" +
+                                                "\"https://msft.sts.microsoft.com/adfs/services/trust/mex\"," +
+                                                "\"federation_active_auth_url\":\"https://msft.sts.microsoft.com/adfs/services/trust/2005/usernamemixed\"" +
+                                                ",\"cloud_instance_name\":\"login.microsoftonline.com\"}")
+                },
+                QueryParams = new Dictionary<string, string>()
+                {
+                    {"api-version", "1.0"}
+                }
+            });
+
+            HttpMessageHandlerFactory.AddMockHandler(new MockHttpMessageHandler("https://msft.sts.microsoft.com/adfs/services/trust/mex")
+            {
+                Method = HttpMethod.Get,
+                ResponseMessage = new HttpResponseMessage(HttpStatusCode.NotFound)
+                {
+                    Content = new StringContent(File.ReadAllText("TestMex.xml"))
+                }
+            });
+
+            string emptyPassword = null;
+            UserPasswordCredential userPasswordCredential = new UserPasswordCredential(TestConstants.DefaultDisplayableId, emptyPassword);
+
+            // Call acquire token
+            var result = AssertException.TaskThrows<AdalException>(() =>
+            context.AcquireTokenAsync(TestConstants.DefaultResource, TestConstants.DefaultClientId,
+                userPasswordCredential));
+
+            // Check inner exception
+            Assert.AreEqual(" Response status code does not indicate success: 404 (NotFound).", result.InnerException.Message);
+
+            // All mocks are consumed
+            Assert.AreEqual(0, HttpMessageHandlerFactory.MockHandlersCount());
+        }
+
+        [TestMethod]
+        [Description("Test for AcquireToken with valid token in cache")]
+        public async Task AcquireTokenWithValidTokenInCache_ReturnsCachedTokenTestAsync()
```

Comment:
this is a managed user test case form the code. You should be using UserCredential class and mock call should be failing with 400 saying "no network call is expected". this is to ensure that no network call actually gets made and if it does then we do know how to fix it

Reply:
Thanks for the explanation on this.